### PR TITLE
fix: lower case header will get better performance

### PR DESCRIPTION
**Pull request by fengmk2**
> Each header field consists of a name followed by a colon (":") and the field value. Field names are case-insensitive.

https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.2

### CODE REVIEW

**Comment by fengmk2** [comment]
我再提交一个性能对比

File: lib/middlewares/csrf.js [hunk 1]
```diff
@@ -3,16 +3,17 @@
 const debug = require('debug')('egg-security:csrf');
 const utils = require('../utils');
 
-module.exports = function(options) {
+module.exports = options => {
   return function* csrf(next) {
     if (this.isAjax) {
```
Comment (dead-horse) [inline lib/middlewares/csrf.js:8]: 这里为什么加了  isAjax 的判断？ isAjax 感觉不太严谨啊，会不会有些 post 请求就被忽略了？

Comment (fengmk2) [reply lib/middlewares/csrf.js:8]: 这是之前的逻辑，要修复单独提 pr

Comment (dead-horse) [reply lib/middlewares/csrf.js:8]: @jtyjty99999 确认一下？

**Comment by fengmk2** [comment]
10x 性能提升..

``` bash
setHeader() Benchmark
  node version: v6.4.0, date: Tue Aug 30 2016 10:53:16 GMT+0800 (CST)
  Starting...
  4 tests completed.

  set()                                      x    433,595 ops/sec ±5.37% (79 runs sampled)
  set() with lower case name                 x  3,895,403 ops/sec ±1.61% (86 runs sampled)
  setWithoutLowerCase()                      x 10,567,733 ops/sec ±1.69% (86 runs sampled)
  setWithoutLowerCase() with lower case name x 10,217,612 ops/sec ±1.22% (87 runs sampled)
```

**Pull request merged**